### ISSUE TITLE
missing page should return 404

### ISSUE THREAD

**Issue by newgene** [SmartAPI/smartAPI#154]
Currently an invalid URL like https://dev.smart-api.info/somepage return correct customized 404 page, but the status code is still 200, ideally, status code should be 404 instead.

**Comment by marcodarko** [comment]
In a Single Page Application, handling 404 errors can be a bit different since the routing is usually handled client-side by frameworks like Vue, React, or Angular and requests errors are not set by these frameworks. This has to be done on the server side probably on nginx, but there lies the complication. Since all the routes the app can handle live on frontend code we would need a way to share that with nginx but since some of the routes are dynamic eg. smartAPi UI pages or documentation pages, that would make it difficult to do.  I will need to do more research on this or maybe we need to move SmartAPI to be a server side rendered app like DDE. If using SSR (e.g., with Nuxt.js for Vue), you can set the 404 status code directly on the server and I assume a little tweaking of the nginx rules to pass the correct code. 
TLDR: The SPA cannot change the status code since the page has already been served, we need to change the strategy to detect this at the server side to achieve this.

**Comment by newgene** [comment]
@marcodarko If this is the normal behavior for a client-side SPA, it should be fine. Let's close this one now.
